packages/big-design/src/components/Modal/Modal.tsx

```diff
@@ -18,170 +18,146 @@ import {
+        ),
+      [header, headerUniqueId],
+    );
+
+    const renderedActions = useMemo(
+      () =>
+        Array.isArray(actions) && (
+          <StyledModalActions justifyContent="flex-end">
+            {actions.map(({ text, ...props }, index) => (
+              <Button key={index} {...props}>
+                {text}
+              </Button>
+            ))}
+          </StyledModalActions>
+        ),
+      [actions],
+    );
+
+    const renderedContent = (
       <StyledModal
         backdrop={backdrop}
-        onClick={this.onClickAway}
-        onKeyDown={this.onKeyDown}
-        ref={this.modalRef}
+        onClick={onClickAway}
+        onKeyDown={onKeyDown}
+        ref={modalRef}
         variant={variant}
         tabIndex={-1}
       >
-        <StyledModalContent variant={variant} aria-labelledby={this.headerUniqueId} flexDirection="column">
-          {this.renderClose()}
-          {this.renderHeader()}
+        <StyledModalContent variant={variant} aria-labelledby={headerUniqueId} flexDirection="column">
+          {renderedClose}
+          {renderedHeader}
           <StyledModalBody>{children}</StyledModalBody>
-          {this.renderActions()}
+          {renderedActions}
         </StyledModalContent>
       </StyledModal>
     );
 
-    return createPortal(modalContent, modalContainer);
-  }
-
-  private renderClose() {
-    const { onClose, variant } = this.props;
-
-    return (
-      variant === 'modal' && (
-        <StyledModalClose>
-          <Button onClick={onClose} iconOnly={<CloseIcon title="Close" />} variant="subtle" />
-        </StyledModalClose>
-      )
-    );
-  }
-
-  private renderHeader() {
-    const { header } = this.props;
-
-    return (
-      header &&
-      typeof header === 'string' && (
-        <StyledModalHeader id={this.headerUniqueId}>
-          <H2 margin="none">{header}</H2>
-        </StyledModalHeader>
-      )
-    );
-  }
-
-  private renderActions() {
-    const { actions } = this.props;
-
-    return (
-      Array.isArray(actions) && (
-        <StyledModalActions justifyContent="flex-end">
-          {actions.map(({ text, ...props }, index) => (
-            <Button key={index} {...props}>
-              {text}
-            </Button>
-          ))}
-        </StyledModalActions>
-      )
-    );
-  }
-
-  private returnFocus = () => {
-    const previousFocus = this.previousFocus as HTMLElement;
-
-    if (previousFocus && typeof previousFocus.focus === 'function') {
-      previousFocus.focus();
-    }
-  };
-
-  private onClickAway = (event: React.MouseEvent) => {
-    if (this.props.closeOnClickOutside && this.modalRef.current === event.target) {
-      this.props.onClose();
-    }
-  };
-
-  private onKeyDown = (event: React.KeyboardEvent) => {
-    if (this.props.closeOnEscKey && event.key === 'Escape') {
-      this.props.onClose();
-    }
-  };
-}
+    return !(isOpen && modalContainer) ? null : createPortal(renderedContent, modalContainer);
```

Comment:
What is the difference between this and `return isOpen && modalContainer ? createPortal(renderedContent, modalContainer) : null` ?

Reply:
Good catch, I can change it over.